### PR TITLE
GitHub URI Parameters

### PR DESCRIPTION
**Pull request by tdonaworth**
* Introduced two parameters to set custom GitHub Enterprise URIs: -github-api-uri, -github-graphql-uri
* Updated documentation to reflect the new parameters

Parameters default to Github.com URIs and are optional.

### CODE REVIEW

File: docs/run.md [hunk 1]
```diff
@@ -78,6 +78,24 @@ The following ingestors need to run first and in this particular order:
 
 Order doesn't matter for other ingestors.
 
+#### Github Enterprise
```
Comment (serain) [inline docs/run.md:81]: nitpick: GitHub

File: cmd/main.go [hunk 2]
```diff
@@ -13,12 +13,14 @@ import (
...
-	neo4jUser     string
-	neo4jPassword string
-	session       string
+	debug            bool
+	organization     string
+	neo4jURI         string
+	neo4jUser        string
+	neo4jPassword    string
+	session          string
+	githubApiURI     string
```
Comment (serain) [inline cmd/main.go:22]: Since these parameters are for the GitHub command, could we shift them below? next to `githubToken`?

File: cmd/main.go [hunk 3]
```diff
@@ -108,6 +110,8 @@ func setupCommonFlags() {
 		fs.StringVar(&neo4jURI, "neo4j-uri", "neo4j://localhost:7687", "The Neo4j URI.")
 		fs.StringVar(&neo4jUser, "neo4j-user", "neo4j", "The Neo4j user.")
 		fs.StringVar(&neo4jPassword, "neo4j-password", "", "The Neo4j password.")
+		fs.StringVar(&githubApiURI, "github-api-uri","https://api.github.com", "The target GitHub API URI, defaults to https://api.github.com; Change if targeting Github Enterprise.")
```
Comment (serain) [inline cmd/main.go:113]: since these are only relevant for the GitHub command they should be subcommand flags (see `githubToken` above)

**Review by serain — CHANGES_REQUESTED** [review]
hey @tdonaworth , thanks so much for adding support for this

have you been able to successfully test `gitoops` against a GH Enterprise? I didn't have anyone in my network who could test this, so would be curious to know if it works alright :) 

I've left a few minor comments regarding this PR.

File: docs/run.md (hunk 1 above)
Comment (tdonaworth) [reply docs/run.md:81]: > hey @tdonaworth , thanks so much for adding support for this
> 
> have you been able to successfully test `gitoops` against a self-hosted GH Enterprise? I didn't have anyone in my network who could test this, so would be curious to know if it works alright :)
> 
> I've left a few minor comments regarding this PR.

Yes, I've been running it against a client's GitHub Enterprise instance - I first just hard-coded the values to see if everything else would work; when it did, I refactored what you see here. I'll review the comments and look to make some changes - but Go is my primary language :(

File: cmd/main.go (hunk 3 above)
Comment (tdonaworth) [reply cmd/main.go]: latest commit should fix this - let me know if this is what you were referring to.

File: cmd/main.go (hunk 2 above)
Comment (tdonaworth) [reply cmd/main.go]: fixed

**Comment by serain** [comment]
Thanks!

I've made some changes, mostly cosmetic and formatting.

Let me know if you're happy for me to merge.

**Review by serain — COMMENTED** [review]
Thanks!

I've made some cosmetic and formatting changes.

Let me know if you're happy for me to merge.

**Comment by tdonaworth** [comment]
> Thanks!
> 
> I've made some cosmetic and formatting changes.
> 
> Let me know if you're happy for me to merge.

No issues here!

**Pull request merged**